Plugins/Json/src/GridJsonConverter.cpp

```
@@ -43,130 +44,256 @@ nlohmann::json Acts::AxisJsonConverter::toJsonDetray(const IAxis& ia) {
+        globalToGridLocal = decodeSubspace<binZ>(jGlobalToGridLocal);
+        break;
+      case binR:
+        globalToGridLocal = decodeSubspace<binR>(jGlobalToGridLocal);
+        break;
+      case binPhi:
+        globalToGridLocal = decodeSubspace<binPhi>(jGlobalToGridLocal);
+        break;
+      case binEta:
+        globalToGridLocal = decodeSubspace<binEta>(jGlobalToGridLocal);
+        break;
+      default:
+        break;
+    }
+  }
+
+  // Switch and fill for 2D
+  if (accessors.size() == 2u) {
+    if (accessors == std::vector<BinningValue>{binX, binY}) {
+      globalToGridLocal = decodeSubspace<binX, binY>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binY, binX}) {
+      globalToGridLocal = decodeSubspace<binY, binX>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binX, binZ}) {
+      globalToGridLocal = decodeSubspace<binX, binZ>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binX}) {
+      globalToGridLocal = decodeSubspace<binZ, binX>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binY, binZ}) {
+      globalToGridLocal = decodeSubspace<binY, binZ>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binY}) {
+      globalToGridLocal = decodeSubspace<binZ, binY>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binR, binPhi}) {
+      globalToGridLocal = decodeSubspace<binR, binPhi>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binPhi, binR}) {
+      globalToGridLocal = decodeSubspace<binPhi, binR>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binZ, binPhi}) {
+      globalToGridLocal = decodeSubspace<binZ, binPhi>(jGlobalToGridLocal);
+    } else if (accessors == std::vector<BinningValue>{binPhi, binZ}) {
+      globalToGridLocal = decodeSubspace<binPhi, binZ>(jGlobalToGridLocal);
+    }
```

```suggestion
    }
    // else globalToGridLocal = nullptr;
```

---

It should be set to nullptr already - is this just for readability?

---

Yes, just for readability, since it is set quite a few lines above.

I thought about adding a `throw` instead of the `break` but just returning `nullptr` could be fine as well.